tests/Microsoft.Identity.Test.Unit/PublicApiTests/CacheKeyExtensionTests.cs

```diff
@@ -0,0 +1,207 @@
+
+            //Ensure that the order of the keys does not matter
+            result = await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                                .WithAdditionalCacheKeyComponents(_additionalCacheKeys3)
+                                .ExecuteAsync(CancellationToken.None)
+                                .ConfigureAwait(false);
+
+            Assert.IsNotNull(result);
+            Assert.AreEqual("header.payload.signature", result.AccessToken);
+            Assert.AreEqual(1, app.AppTokenCacheInternal.Accessor.GetAllAccessTokens().Count);
+            Assert.AreEqual(TokenSource.Cache, result.AuthenticationResultMetadata.TokenSource);
+
+            //Ensure that tokens are not ovverriden
+            httpManager.AddMockHandlerSuccessfulClientCredentialTokenResponseMessage();
+
+            result = await app.AcquireTokenForClient(TestConstants.s_scopeForAnotherResource)
+                                .WithAdditionalCacheKeyComponents(_additionalCacheKeys2)
+                                .ExecuteAsync(CancellationToken.None)
+                                .ConfigureAwait(false);
+
+            Assert.IsNotNull(result);
+            Assert.AreEqual("header.payload.signature", result.AccessToken);
+            Assert.AreEqual(2, app.AppTokenCacheInternal.Accessor.GetAllAccessTokens().Count);
+            Assert.AreEqual(TokenSource.IdentityProvider, result.AuthenticationResultMetadata.TokenSource);
+        }
+
+        [TestMethod]
+        public async Task CacheExtNullOrEmptyTestAsync()
+        {
+            using (var httpManager = new MockHttpManager())
+            {
+                var app = ConfidentialClientApplicationBuilder.Create(TestConstants.ClientId)
+                                                              .WithAuthority(new Uri(ClientApplicationBase.DefaultAuthority), true)
+                                                              .WithRedirectUri(TestConstants.RedirectUri)
+                                                              .WithClientSecret(TestConstants.ClientSecret)
+                                                              .WithHttpManager(httpManager)
+                                                              .WithExperimentalFeatures()
+                                                              .BuildConcrete();
+
+                var exception = await Assert.ThrowsExceptionAsync<MsalClientException>(async () =>
+                {
+                    await app.AcquireTokenForClient(TestConstants.s_scope.ToArray())
+                                            .WithAdditionalCacheKeyComponents(null)
+                                            .ExecuteAsync(CancellationToken.None)
+                                            .ConfigureAwait(false);
+                }).ConfigureAwait(false);
+
+                Assert.AreEqual(MsalErrorMessage.CryptographicError, exception.Message);
```

Comment:
That's very poor devex.

Reply:
I think this is actually incorrect to begin with, this was supposed to be an `ArgumentNull` exception instead of a CryptographicError. Probably pressed ctrl z too many times.